### ISSUE TITLE
Possible key generation incompatibility with ring

### ISSUE THREAD

**Issue by shutton** [Devolutions/picky-rs#67]
I was working on a code snippet to generate a fresh key pair and hand it to tokio-rustls (which uses [ring](https://crates.io/crates/ring) for its crypto work).  However, the keys generated by PrivateKey::generate_rsa don't seem to make it happy.  If you bypass rustls, as in this test program:
``` rust
fn main() {
    let server_private_key = picky::key::PrivateKey::generate_rsa(2048).unwrap();
    let res = ring::signature::RsaKeyPair::from_pkcs8(&server_private_key.to_pkcs8().unwrap());
    eprintln!("Result = {:?}", res);
}
```
you'll get back `Err(KeyRejected("InconsistentComponents"))`

Unfortunately, `ring` uses this error for *many* conditions, but I tracked it down to this particular check in `limb.rs:parse_big_endian_and_pad_consttime()`:
``` rust
    if num_encoded_limbs > result.len() {
        return Err(error::Unspecified);
    }
```
It fails due to the encoded limbs being 32, but result.len() being 16.

I'm at a bit of a loss and was hoping this might ring a bell (no pun intended) with the picky development team.  It's presumably an issue with either the `rsa` crate (which picky uses) or `ring`, but I'm not positive which direction to head first.

**Comment by shutton** [comment]
I managed to find a way to generate RSA private keys that work with ring.  The issue appears to be related to PKCS#8 encoding.  When an RSA key is exported to PKCS#8, it generates a payload that appears to have a mismatch in the expected key type.  Ring complains that the algorithms are mismatched (though I'm not entirely certain where it's getting the information), seeing that RSA_ENCRYPTION (OID 1.2.840.113549.1.1.1) != SHA1_WITH_RSA_ENCRYPTION (OID 1.2.840.113549.1.1.5).

I was able to get keys compatible with ring using the following recipe (apologies for using `anyhow` to wrap the error):
```
fn generate_private_key(bits: usize) -> Result<picky::key::PrivateKey> {
    let key = RSAPrivateKey::new(&mut OsRng, bits).unwrap();
    let key = rsa_export::pkcs1::private_key(&key)
        .map_err(|_| anyhow!("unable to export private key"))?;
    Ok(picky::key::PrivateKey::from_rsa_der(&key)?)
}
```

This was my alternative to using `picky::key::PrivateKey::generate_rsa()`

**Comment by bcortier-devolutions** [comment]
I found the issue. Looks like we were indeed generating a bad exponent in our pkcs8 and thus "InconsistentComponents".
These exponents are used to speed up decryption (`ring` uses them). However they are currently ignored by `rsa` crate so we never stumbled on the issue up to this day.

That's how one computes the exponents:
```
dp = d mod (p - 1)
dq = q mod (q - 1)
```
But somehow we were doing `dq = q` :facepalm: 

I added a unit test based on your snipped above to make sure we never break again like this.
